crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -929,6 +929,133 @@ impl Measured2d for Annulus {
     }
 }
 
+/// A rhombus primitive, also known as a diamond shape.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[doc(alias = "Diamond")]
+pub struct Rhombus {
+    /// Size of the horizontal and vertical diagonals of the rhombus
+    pub half_diagonals: Vec2,
+}
+impl Primitive2d for Rhombus {}
+
+impl Default for Rhombus {
+    /// Returns the default [`Rhombus`] with a half-horizontal and half-vertical diagonal of `0.5`.
+    fn default() -> Self {
+        Self {
+            half_diagonals: Vec2::splat(0.5),
+        }
+    }
+}
+
+impl Rhombus {
+    /// Create a new `Rhombus` from a vertical and horizontal diagonal sizes.
+    #[inline(always)]
+    pub fn new(horizontal_diagonal: f32, vertical_diagonal: f32) -> Self {
+        Self {
+            half_diagonals: Vec2::new(horizontal_diagonal / 2.0, vertical_diagonal / 2.0),
+        }
+    }
+
+    /// Create a new `Rhombus` from a side length with all inner angles equal.
+    #[inline(always)]
+    pub fn from_side(side: f32) -> Self {
+        Self {
+            half_diagonals: Vec2::new(side.hypot(side) / 2.0, side.hypot(side) / 2.0),
```

Comment:
```suggestion
            half_diagonals: Vec2::splat(side * std::f32::consts::FRAC_1SQRT_2),
```

Reply:
When we tried this solution the results were just some decimal places changed (for example, instead of 1.0 it was 0.999994), so, in our last commit, we decided to use the `splat` but with the `hypot` so the precision is as high as possible.

Reply:
That is odd... I guess it's not a huge deal in a constructor. Btw `0.999994` is pretty much within f32 precision (about 7 decimal places) IIRC.